legate/core/context.py

```diff
@@ -18,6 +18,8 @@
 
 import numpy as np
 
+import legate.core.types as ty
```

Comment:
I think if this is used only in type annotations it needs to go below in the guarded `if TYPE_CHECKING` imports.